ui/apps/platform/src/hooks/useFetchNamespaceDeployments.ts

```diff
@@ -0,0 +1,66 @@
+import { useState, useEffect } from 'react';
+import { gql, useQuery } from '@apollo/client';
+
+import queryService from 'utils/queryService';
+
+type Deployment = {
+    id: string;
+    name: string;
+};
+export type NamespaceWithDeployments = {
+    metadata: {
+        id: string;
+        name: string;
+    };
+    deployments: Deployment[];
+};
+type DeploymentResponse = {
+    results: NamespaceWithDeployments[];
+};
+
+const DEPLOYMENTS_FOR_NAMESPACE_QUERY = gql`
+    query getNamespaceDeployments($query: String!) {
+        results: namespaces(query: $query) {
+            metadata {
+                name
+                id
+            }
+            deployments {
+                name
+                id
+            }
+        }
+    }
+`;
+
+function useFetchNamespaceDeployments(selectedNamespaceIds: string[]) {
```

Comment:
Should we keep this hook within the `NetworkGraph` directory for now until we know it's needed in other areas?

Reply:
I thought about that, but I erred on the side of keeping it with the other API helper hooks, because I've noticed a few times in the past devs writing basically the same hook for a Container, because they didn't realize someone else had needed it and added it to another Container.